docs/deployment-ux.md

```diff
@@ -0,0 +1,410 @@
+Blox Deployments User Experience
+================================
+
+Use Cases
+---------
+
+The following are typical use cases for Deployments:
+
+-   Create an Environment to deploy a new environment revision. The
+    Deployment creates Tasks in the background. Check the status of the
+    deployment to see if it succeeds or not.
+-   Declare the new state of the Environment by updating it, then deploy
+    it by calling startDeployment. A new Environment revision is created
+    and the Deployment manages moving the Tasks from the old revision to
+    the new one at a controlled rate.
+-   Rollback to an earlier Enviornment revision if the current state of
+    the Environment is not stable. Each rollback updates the revision of
+    the environment (it does not simply revert the environment
+    revision).
+-   (not applicable for Daemon) Scale up the Deployment to facilitate
+    more load.
+-   Pause the Deployment to apply multiple fixes to its Environment and
+    then resume it to start a new deployment.
+-   Use the status of the Deployment as an indicator that a rollout has
+    become stuck.
+
+Creating and deploying an Environment
+-------------------------------------
+
+In order to create a deployment, you first have to create and
+environment that declares what should be running, where, and how the
+deployment should be controlled.
+
+``` {.shell}
+$ aws ecs create-environment --yaml-file <<EOF
+Name: "SomeEnvironment/Prod"
+TaskDefinition: "log-daemon:1"
+Cluster: "default"
+InstanceGroup: 
+  Query: "attribute:stack == prod"
+DeploymentType: Daemon
+DeploymentConfiguration:
+  Method: "Replace" # (or "Surge")
+  MinHealthyPercent: "100"
+EOF
+{ "EnvironmentRevision": 1 }
+```
+
+At this point, nothing has actually happened to your cluster. You can
+see this by showing your environment's status:
+
+``` {.shell}
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS     DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
```

Comment:
I copied the table headers from `k8s`s format. Up-to-date and Available don't yet have analogous concepts in Blox, so I'll remove them from all examples.